lib/pymedphys/tests/experimental/pseudonymisation/test_pseudonymisation.py

```diff
@@ -81,6 +81,29 @@ def test_pseudonymise_convenience_api():
                 _assert_values_changed_and_not_hardcoded(input_file, pseudo_file)
 
 
+@pytest.mark.pydicom
+def test_identifier_with_DS_vr():
+    # PatientWeight is of type DS
+    # Because the strategy is only applied when the identifier is found
+    # in the dataset, the defect only surfaced in that circumstance
+    replacement_strategy = pseudonymisation_api.pseudonymisation_dispatch
+    logging.info("Using pseudonymisation strategy")
+    identifying_keywords = ["PatientID", "PatientWeight"]
+    ds_input = pydicom.Dataset()
+    ds_input.PatientID = "ABC123"
+    ds_input.PatientWeight = "73.2"
```

Comment:
Not for this PR, but it might be of significant value to find these sorts of edge cases by using the Hypothesis testing library.

Reply:
I'll investigate using hypothesis for further testing of the pseudonymisation strategy.